packages/quantic/force-app/main/default/lwc/quanticSourceCitations/quanticSourceCitations.js

```diff
@@ -0,0 +1,50 @@
+import learnMore from '@salesforce/label/c.quantic_LearnMore';
+import {LightningElement, api} from 'lwc';
+
+/** @typedef {import("coveo").GeneratedAnswerCitation} GeneratedAnswerCitation */
+
+/**
+ * The `QuanticSourceCitations` component renders the citations used to generate the answer in the quantic generated answer component.
+ * @category Internal
+ * @example
+ * <c-quantic-source-citations citations={citations} citation-click-handler={citationClickHandler}></c-quantic-source-citations>
+ */
+export default class QuanticSourceCitations extends LightningElement {
+  labels = {
+    learnMore,
+  };
+
+  /**
+   * The citations used in the generated answer.
+   * @api
+   * @type {GeneratedAnswerCitation[]}
+   */
+  @api citations;
+  /**
+   * The function to be executed when a citation is clicked.
+   * @api
+   * @type {function}
+   */
+  @api citationClickHandler;
+
+  /**
+   * Returns the indexed citations.
+   * @returns {Object}
+   */
+  get indexedCitations() {
+    return this.citations.map((citation, index) => ({
+      ...citation,
+      index: index + 1,
+      handleCitationClick: () => {
+        this.citationClickHandler?.(citation.id);
```

Comment:
Ohhh nevermind the above comment, I didn't notice we were keeping the original click on the anchor tag to open in a new tab.

We'll have an issue then if the result is a Salesforce source. As it will try to open in a new tab the source url instead of using the Salesforce navigation mixin to navigate to the result.

We should update this component then I thought only the parent component would control what happens on a click.

Reply:
Take a look at the `quanticResultLink` https://github.com/coveo/ui-kit/blob/02e4242961bfd34609ed88c953eefe1924b81ba6/packages/quantic/force-app/main/default/lwc/quanticResultLink/quanticResultLink.js#L104

Reply:
how would we make sure the citation is Salesforce if they dont have sfids in their JSON?

Reply:
Yea I just realized this would become a problem... I guess it can't really be helped unless we start returning the sfid if the field is present.

Probably a future improvement. I'll talk to Nathan today about it, but for now consider it not supported 👍